lib/spotlight/dor/indexer.rb

```diff
@@ -189,19 +192,20 @@ def add_folder_name(sdb, solr_doc)
 
       # search for configured full text files, and if found, add them to the full text (whole document) solr field
       def add_object_full_text(sdb, solr_doc)
-        object_level_full_text_urls(sdb).each do |file_url|
-          # append content from each file to the field, creating field if it doesn't exist yet
-          # ruby note: the construct below allows us to append a string to a variable that starts out as nil
-          # because .to_s of nil is empty string
-          solr_doc['full_text_tesim'] = solr_doc['full_text_tesim'].to_s + get_file_content(file_url)
+        full_text_urls = object_level_full_text_urls(sdb)
+        full_text_field_name = 'full_text_tesim'
+        return if full_text_urls.size == 0
+        solr_doc[full_text_field_name] = []
+        full_text_urls.each do |file_url|
```

Comment:
good suggestion - made some tweaks to your suggestion so that the code still returns nil for the field when no docs exist